### PR TITLE
🔥 feat: Add HeaderLimit option to BasicAuth middleware

### PR DESCRIPTION
**Pull request by gaby**
## Summary
- add `HeaderLimit` option to BasicAuth middleware
- enforce authorization header length in middleware
- document `HeaderLimit` in docs
- note the change in what's new
- test header limit handling
- add unit tests for empty or whitespace-only auth headers

### CODE REVIEW

File: middleware/basicauth/basicauth.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func New(config Config) fiber.Handler {
 
 		// Get authorization header and ensure it matches the Basic scheme
 		auth := utils.Trim(c.Get(fiber.HeaderAuthorization), ' ')
-		if auth == "" {
+		if auth == "" || len(auth) > cfg.HeaderLimit {
 			return cfg.Unauthorized(c)
 		}
```
[the remark this answers is not in the document]
Comment (gaby) [reply middleware/basicauth/basicauth.go:39]: The trimmed header value becomes the new value though?
